src/main/java/org/hobbit/core/containerservice/DirectContainerCreator.java

```diff
@@ -0,0 +1,195 @@
+     */
+	public void createTaskGenerators(String taskGeneratorImageName, int numberOfTaskGenerators,
+            String[] envVariables, AbstractCommandReceivingComponent dummyComponent) {
+		this.directComponent = dummyComponent;
+		createGenerator(taskGeneratorImageName, numberOfTaskGenerators, envVariables, abstractBenchmarkController.getTaskGenContainerIds());
+    }
+	
+	/**
+     * Creates the evaluate storage using the given image name and environment
+     * variables.
+     *
+     * @param evalStorageImageName
+     *            name of the evaluation storage image
+     * @param envVariables
+     *            environment variables that should be given to the component
+     */
+	
+	public void createEvaluationStorage(String evalStorageImageName, String[] envVariables,
+			AbstractCommandReceivingComponent dummyComponent) {
+		this.directComponent = dummyComponent;
+		abstractBenchmarkController.setEvalStoreContainerId(abstractBenchmarkController.createContainer(evalStorageImageName, Constants.CONTAINER_TYPE_DATABASE, envVariables));
+        if (abstractBenchmarkController.getEvalStoreContainerId() == null) {
+            String errorMsg = "Couldn't create evaluation storage. Aborting.";
+            LOGGER.error(errorMsg);
+            throw new IllegalStateException(errorMsg);
+        }
+    }
+
+	/**
+     * Internal method for creating generator components.
+     *
+     * @param generatorImageName
+     *            name of the generator Docker image
+     * @param numberOfGenerators
+     *            number of generators that should be created
+     * @param envVariables
+     *            environment variables for the task generators
+     * @param generatorIds
+     *            set of generator container names
+     */
+	public void createGenerator(String generatorImageName, int numberOfGenerators, String[] envVariables,
+            Set<String> generatorIds) {
+		try {
+			String containerId;
+			String variables[] = envVariables != null ? Arrays.copyOf(envVariables, envVariables.length + 2)
+					: new String[2];
+			// NOTE: Count only includes generators created within this method call.
+			variables[variables.length - 2] = Constants.GENERATOR_COUNT_KEY + "=" + numberOfGenerators;
+			for (int i = 0; i < numberOfGenerators; ++i) {
+				// At the start generatorIds is empty, and new generators are added to it immediately.
+				// Current size of that set is used to make IDs for new generators.
+				variables[variables.length - 1] = Constants.GENERATOR_ID_KEY + "=" + generatorIds.size();
+				containerId = createContainer(generatorImageName, null, envVariables, null).get();// createContainer(generatorImageName, variables);
+				if (containerId != null) {
+					generatorIds.add(containerId);
+				} else {
+					String errorMsg = "Couldn't create generator component. Aborting.";
+					LOGGER.error(errorMsg);
+					throw new IllegalStateException(errorMsg);
+				}
+			}
+		}catch(Exception e) {
+			
```

Comment:
Ignoring an exception is not a good idea. If there is an error, it should be logged. Apart from that, the components reacted on a failed container creation in the previous implementations. This does not seem to be the case in this implementation. However, that is not good. A component needs to be able to recognize that the creation of a container has failed to react accordingly.